src/shared/components/fusionTable/FusionTable.tsx

```diff
@@ -0,0 +1,249 @@
+/**
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+import * as _ from "lodash";
+import * as React from "react";
+import { observer } from "mobx-react";
+import { observable, computed } from "mobx";
+import { default as LazyMobXTable, Column, SortDirection } from "shared/components/lazyMobXTable/LazyMobXTable";
+import { IFusionTableProps } from './IFusionTableProps';
+import { StructuralVariant } from '../../api/generated/CBioPortalAPI';
+
+/**
+ * Fusion table column types
+ */
+export enum FusionTableColumnType {
+    SAMPLE_ID,
+
+    SITE1_ENTREZ_GENE_ID,
+    SITE1_HUGO_SYMBOL,
+    SITE1_ENSEMBL_TRANSCRIPT_ID,
+    SITE1_EXON,
+    SITE1_CHROMOSOME,
+    SITE1_POSITION,
+    SITE1_DESCRIPTION,
+
+    SITE2_ENTREZ_GENE_ID,
+    SITE2_HUGO_SYMBOL,
+    SITE2_ENSEMBL_TRANSCRIPT_ID,
+    SITE2_EXON,
+    SITE2_CHROMOSOME,
+    SITE2_POSITION,
+    SITE2_DESCRIPTION,
+    SITE2_EFFECT_ON_FRAME,
+
+    NCBI_BUILD,
+    DNA_SUPPORT,
+    RNA_SUPPORT,
+
+    NORMAL_READ_COUNT,
+    TUMOR_READ_COUNT,
+
+    NORMAL_VARIANT_COUNT,
+    TUMOR_VARIANT_COUNT,
+
+    NORMAL_PAIRED_END_READ_COUNT,
+    TUMOR_PAIRED_END_READ_COUNT,
+
+    NORMAL_SPLIT_READ_COUNT,
+    TUMOR_SPLIT_READ_COUNT,
+
+    BREAKPOINT_TYPE,
+    ANNOTATION,
+    CENTER,
+    CONNECTION_TYPE,
+    EVENT_INFO,
+    VARIANT_CLASS,
+    LENGTH,
+    COMMENTS,
+    EXTERNAL_ANNOTATION,
+
+    DRIVER_FILTER,
+    DRIVER_FILTER_ANNOTATION,
+    DRIVER_TIERS_FILTER,
+    DRIVER_TIERS_FILTER_ANNOTATION,
+}
+
+
+export const FusionTableColumnLabels = [
+    { columnType: FusionTableColumnType.SAMPLE_ID, label: 'Sample Id', attribute: 'sampleId', visible: false},
```

Comment:
could there be something wrong with the code structure if the same list of fields is being repeated 4 times? I personally think it is quite odd to have to basically repeat this list again 4 times.

Reply:
So far what I see from how they did it in Mutation table, we want to map column type itself which resembles db names , labels (what you see in the ui) and the as object attributes from backend